snuba/datasets/storages/transactions.py

```diff
@@ -67,6 +67,10 @@
         ("message_timestamp", DateTime()),
         ("retention_days", UInt(16)),
         ("deleted", UInt(8)),
+        ("type", String(Modifiers(readonly=True))),
+        ("message", String(Modifiers(readonly=True))),
+        ("title", String(Modifiers(readonly=True))),
+        ("timestamp", DateTime(Modifiers(readonly=True))),
```

Comment:
Why do we need to add these to the storage?  The transaction storage should not need these fiedls with the new approach.

Reply:
The storage schema columns must always be in sync with the schema generated by the migration - we have a test that checks for schema differences that will fail if it isn't the case. I'm not sure that we want to relax this requirement just for this.

Reply:
So we still need to add those column in both tables to make the schema match ?

Reply:
Yes - we can't have merge table with columns that aren't present in both underlying tables

Reply:
Ah ok, now I read it better, `if(_table = 'transactions_local', message, transaction_name)` is an alternative solution to try out later.

Reply:
What about adding a modifier to the migration column to signal that such column should not be visible in the related schema and thus it would be skipped when checking the schema correspond? This way we would not have to add these columns in the schema (we add them only in the migration) and  thus they would be hidden to the query processing pipeline.

Reply:
I think it actually should be visible in the related schema since the Discover entity will use this column when the transaction storage is picked and it makes sense to be visible to the query processing pipeline when it is being selected. (Both DiscoverTransactions storage and Transactions storage share the same schema, and will eventually be merged together. Transactions are also only queried through Discover anyway where the column is always relevant.). I don't think there's too much benefit in attempting to hide the fact that this column exists.

Reply:
mh, ok.